libsolidity/codegen/CompilerContext.cpp

```diff
@@ -312,8 +313,23 @@ void CompilerContext::appendInlineAssembly(
 	ErrorReporter errorReporter(errors);
 	auto scanner = make_shared<Scanner>(CharStream(_assembly), "--CODEGEN--");
 	auto parserResult = assembly::Parser(errorReporter).parse(scanner);
-	solAssert(parserResult, "Failed to parse inline assembly block.");
-	solAssert(errorReporter.errors().empty(), "Failed to parse inline assembly block.");
+	if (!parserResult || !errorReporter.errors().empty())
+	{
+		cerr << "Error parsing inline assembly block:" << endl;
+		cerr << "------------------ Input: -----------------" << endl;
+		cerr << _assembly << endl;
+		cerr << "------------------ Errors: ----------------" << endl;
+		for (auto const& error: errorReporter.errors())
+			SourceReferenceFormatter::printExceptionInformation(
+				cerr,
+				*error,
+				(error->type() == Error::Type::Warning) ? "Warning" : "Error",
+				[&](string const&) -> Scanner const& { return *scanner; }
+			);
+		cerr << "-------------------------------------------" << endl;
+
+		solAssert(false, "Failed to parse inline assembly block.");
```

Comment:
I think it would be nicer to include that text in the assert and not on `cerr` as for example soljson shouldn't really touch `cerr`.

Is there a technical limit to the string size in the assert?

Reply:
You are right, I can do that.

Reply:
Also I ran into this assert a couple of times myself :)